mysql-test/suite/tokudb/t/type_date.test

```diff
@@ -207,19 +209,21 @@ DROP TABLE t1,t2;
 # Bug #31928: Search fails on '1000-00-00' date after sql_mode change
 #
 
+SET sql_mode = 'ALLOW_INVALID_DATES';
 CREATE TABLE t1 (a DATE);
 CREATE TABLE t2 (a DATE);
 CREATE INDEX i ON t1 (a);
---error ER_TRUNCATED_WRONG_VALUE
-INSERT INTO t1 VALUES ('1000-00-00'),('1000-00-00');
---error ER_TRUNCATED_WRONG_VALUE
-INSERT INTO t2 VALUES ('1000-00-00'),('1000-00-00');
+INSERT IGNORE INTO t1 VALUES ('1000-00-00'),('1000-00-00');
+INSERT IGNORE INTO t2 VALUES ('1000-00-00'),('1000-00-00');
 SELECT * FROM t1 WHERE a = '1000-00-00';
 SELECT * FROM t2 WHERE a = '1000-00-00';
 SET SQL_MODE=TRADITIONAL;
 replace_column 9 NA;
+--error ER_WRONG_VALUE
```

Comment:
And here same/similar thing, here I think is where we would want to allow invalid values so that we can get the EXPLAIN result rather than an error.